### ISSUE TITLE
[Animation] `animationcomplete` event only fires for first completion

### ISSUE THREAD

**Issue by dsinni** [ngokevin/kframe#137]
When an animation completes, the `animationcomplete` event only fires upon the first completion, and does not fire for any subsequent completions of the same animation.

The `animationbegin` event fires as expected, but `animationcomplete` does not.

I'm experiencing this issue in **A-Frame** `0.7.*` as well as `0.8.0` with **aframe-animation-component** `4.*.*` in **Firefox**, **Chrome**, **Edge** on **Windows 10** (all I've tested).

Demo: https://codepen.io/dansinni/pen/jzBxKa

**Comment by dsinni** [comment]
To clarify, this happens even when looping is disabled, so `animationcomplete` is only good for one fire. :(

**Comment by weddingdj** [comment]
I noticed the same behavior, using A-Frame 0.8.2 with animation component 4.2.0. `animationbegin` fires as expected, but `animationcomplete` does not.

**Comment by krzysztof-grzybek** [comment]
This issue is caused by anime.js unexpected behaviour ([issue link](https://github.com/juliangarnier/anime/issues/277)).

Anime.js simply doesn't trigger `complete` event if it's already completed and then rewinded.

Possible fix:
On each animation start we can remove current animation and create new one.
NOTE: we can't completely remove animation, since anime.js [doesn't provide such method](https://github.com/juliangarnier/anime/issues/188). We can only pause it.

**Comment by yoyTeam** [comment]
to solve problem I reset animation like this
`
element.addEventListener('animationcomplete', function(e){
        this.components[e.detail.name].animation.reset();
        console.log('you can read me');
});
`

**Comment by dsinni** [comment]
Will have to try that. Thanks, @yoyTeam.

**Comment by ngokevin** [comment]
Now fixed with `startEvents` since that restarts and creates new animation each time (and presumably in that case you are triggering the animation start so you can tell already when you are restarting it).

`loop: true` I guess will never complete, and it should fire if it's set to an integer.

**Comment by dsinni** [comment]
Thanks, @ngokevin!

**Comment by janpio** [comment]
Yep, seems to work as expected for `animation` now, but not `animation-timeline`. I created #216 for this.
